lua/weapons/weapon_autoshotgun.lua

```diff
@@ -0,0 +1,120 @@
+SWEP.PrintName = "Auto Shotgun"
+SWEP.Author = TASWeapons.Authors.yousifgaming
+SWEP.Instructions = "Gotta Go Faster"
+SWEP.Purpose = "To play with dingus"
+SWEP.Contact = "Don't Contact me i dont like people"
+
+SWEP.Category = TASWeapons.Category
+SWEP.Spawnable = true
+SWEP.AdminOnly = false
+SWEP.BounceWeaponIcon = false
+SWEP.IconOverride = "autoshotgun/cat.jpg"
+if CLIENT then
+	SWEP.WepSelectIcon = surface.GetTextureID("autoshotgun/cat")
+end
+
+SWEP.Primary.ClipSize = 30
+SWEP.Primary.DefaultClip = 30
+SWEP.Primary.Automatic = true
+SWEP.Primary.Ammo = "Buckshot"
+
+SWEP.Secondary.Ammo = "None"
+SWEP.Secondary.Automatic = true
+
+SWEP.ReloadInProgress = false
+SWEP.AmmoGiven = false
+
+SWEP.Weight = 5
+SWEP.AutoSwitchTo = false
+SWEP.AutoSwitchFrom = false
+
+SWEP.Slot = 1
+SWEP.SlotPos = 0
+SWEP.DrawAmmo = true
+SWEP.DrawCrosshair = true
+
+SWEP.ViewModel = "models/weapons/c_shotgun.mdl"
+SWEP.WorldModel = "models/weapons/w_shotgun.mdl"
+SWEP.UseHands = true
+
+SWEP.ShootSound = Sound("weapons/xm1014/xm1014-1.wav")
+
+resource.AddFile("materials/autoshotgun/cat.vmt")
+resource.AddFile("materials/autoshotgun/cat.vtf")
+resource.AddFile("materials/autoshotgun/cat.jpg")
```

Comment:
These functions do not exist on the client, so they should be wrapped in `if SERVER then`.

Also you shouldn't need to manually `AddFile` the `vtf`:

> For convenience, this function will automatically add any other files that are related to the selected one, and throw an error if it can't find them. **For example, a `.vmt` file will automatically add the `.vtf` with the same name**

https://wiki.facepunch.com/gmod/resource.AddFile